### DISCUSSION TITLE
Adaptive integration over n-cubes and simplices

### DISCUSSION THREAD

**Discussion by Areustle** [nschloe/quadpy#435] [Ideas]
Hello @nschloe, 

Do you have any interest in expanding quadpy to add adaptive cubature algorithms for integrals over n-simplex and n-cube regions? There are a few adaptive algorithms available, mostly from Genz and contemporaries.

https://doi.org/10.1016/0771-050X(80)90039-X
http://www.math.wsu.edu/faculty/genz/papers/cuhre.pdf
https://doi.org/10.1007/BFb0038504

And a C implementation with a python wrapper for rectangular regions:
https://github.com/stevengj/cubature

Curious to hear your thoughts.

**Comment by nschloe** [comment]
> Do you have any interest

Interest yes, time no. I also think this is a missing functionality in quadpy. I've started some work in a branch but haven't finished it yet.

**Comment by Areustle** [comment]
> Interest yes, time no. 

I can fully sympathize with this. I may work on an implementation myself and, if successful, share it here.
